crypto/pem/pem_passwd.c

```diff
@@ -0,0 +1,301 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <assert.h>
+#include <ctype.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <openssl/buf.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+#include <openssl/pem.h>
+#include <openssl/x509.h>
+
+#include "internal.h"
+#include "../internal.h"
+
+# include <signal.h>
+# include <string.h>
+# include <errno.h>
+
+// We support two types of terminal interface:
+// - termios for Linux/Unix
+// - WIN32 Console for Windows
+# if !defined(_WIN32)
+#  define SIGACTION
+#  include <termios.h>
+#  define DEV_TTY "/dev/tty"
+#  define TTY_STRUCT             struct termios
+#  define TTY_FLAGS              c_lflag
+#  define TTY_get(tty,data)      tcgetattr(tty, data)
+#  define TTY_set(tty,data)      tcsetattr(tty, TCSANOW, data)
+# else /* _WIN32 */
+#  include <windows.h>
+# endif
+
+# define NUM_SIG 32
+
+static volatile sig_atomic_t intr_signal;
+static struct CRYPTO_STATIC_MUTEX console_global_mutex = CRYPTO_STATIC_MUTEX_INIT;
+
+# ifdef SIGACTION
+static struct sigaction savsig[NUM_SIG];
+# else
+static void (*savsig[NUM_SIG]) (int);
+# endif
+
+static int read_till_nl(FILE * in);
+static void recsig(int signal);
+static void pushsig(void);
+static void popsig(void);
+
+#if defined(_WIN32)
+  DWORD tty_orig, tty_new;
+#else
+  TTY_STRUCT tty_orig, tty_new;
+#endif
+
+FILE *tty_in, *tty_out;
+int is_a_tty;
+
+
+/* Internal functions to read a string without echoing */
+static int read_till_nl(FILE *in) {
```

Comment:
NP: Rename to something like `discard_line_remainder`? This function's purpose appears to be to discard any remaining characters until it reaches a newline.